client/app.js

```diff
@@ -148,6 +148,183 @@ tie.constant('PREREQ_CHECK_TYPE_BAD_IMPORT', 'badImport');
  * @constant
  */
 tie.constant('PREREQ_CHECK_TYPE_GLOBAL_CODE', 'globalCode');
+tie.constant('PREREQ_CHECK_TYPE_WRONG_LANG', 'wrongLang');
+
+/**
+ * Dictionary of wrong language detection errors and their related information
+ *
+ * @type {{}}
+ */
+tie.constant('WRONG_LANGUAGE_ERRORS', {
```

Comment:
Add Karma test that checks that the internal structure of this is valid. It will be useful when folks add new categories of errors.